### ISSUE TITLE
Incorrectly reports context as text domain on _ex()

### ISSUE THREAD

**Issue by pippinsplugins** [stephenharris/grunt-checktextdomain#11]
The [_ex()](https://developer.wordpress.org/reference/functions/_ex/) function isn't parsed properly. The second parameter, which is the context, is getting read as the text domain, resulting in incorrect error messages:

```
includes/admin/overview/metaboxes/class-metabox-overview-recent-referral-visits.php
[L66]  _e  Incorrect text domain used ("Affiliate column table header")
[L67]  _e  Incorrect text domain used ("URL column table header")
[L68]  _e  Incorrect text domain used ("Converted column table header")
```

That first error is for this:

``` php
_ex( 'Affiliate', 'Affiliate column table header', 'affiliate-wp' )
```

**Comment by stephenharris** [comment]
Are you sure you're not using `_e`? https://github.com/AffiliateWP/AffiliateWP/blob/13c06d17ff82a402df7226248541bd347815a4c8/includes/admin/overview/metaboxes/class-metabox-overview-recent-referral-visits.php#L66 ;)

**Comment by pippinsplugins** [comment]
@stephenharris Yes. https://github.com/AffiliateWP/AffiliateWP/blob/3fa5fcec42a6c1a82b1204cd4d3f4e9593b59d06/includes/admin/overview/metaboxes/class-metabox-overview-most-valuable.php#L74

**Comment by stephenharris** [comment]
That file is `class-metabox-overview-most-valuable.php`, the report is for `class-metabox-overview-recent-referral-visits.php`.

**Comment by pippinsplugins** [comment]
I copied it wrong. Doesn't change the error ;)

**Comment by stephenharris** [comment]
I just cloned AffiliateWP at 3fa5fcec42a6c1a82b1204cd4d3f4e9593b59d06, ran `grunt checktextdomain` and got the following:

```
includes/admin/AFFWP_Plugin_Updater.php
[L196]  __  Incorrect text domain used ("easy-digital-downloads")
[L204]  __  Incorrect text domain used ("easy-digital-downloads")
[L358]  __  Incorrect text domain used ("easy-digital-downloads")
[L358]  __  Incorrect text domain used ("easy-digital-downloads")

includes/admin/overview/metaboxes/class-metabox-overview-recent-referral-visits.php
[L66]  _e  Incorrect text domain used ("Affiliate column table header")
[L67]  _e  Incorrect text domain used ("URL column table header")
[L68]  _e  Incorrect text domain used ("Converted column table header")

includes/admin/overview/metaboxes/class-metabox-overview-recent-referrals.php
[L69]  _e  Incorrect text domain used ("Affiliate column table header")
[L70]  _e  Incorrect text domain used ("Amount column table header")
[L71]  _e  Incorrect text domain used ("Description column table header")

includes/admin/overview/metaboxes/class-metabox-overview-registrations.php
[L67]  _e  Incorrect text domain used ("Affiliate column table header")
[L68]  _e  Incorrect text domain used ("Status column table header")
[L69]  _e  Incorrect text domain used ("Actions column table header")

includes/admin/overview/metaboxes/class-metabox-overview-totals.php
[L54]   _e  Incorrect text domain used ("Paid Earnings column table header")
[L55]   _e  Incorrect text domain used ("Paid Earnings This Month column table header")
[L56]   _e  Incorrect text domain used ("Paid Earnings Today column table header")
[L80]   _e  Incorrect text domain used ("Unpaid Referrals column table header")
[L81]   _e  Incorrect text domain used ("Unpaid Referrals This Month column table header")
[L82]   _e  Incorrect text domain used ("Unpaid Referrals Today column table header")
[L105]  _e  Incorrect text domain used ("Unpaid Earnings column table header")
[L106]  _e  Incorrect text domain used ("Unpaid Earnings This Month")
[L107]  _e  Incorrect text domain used ("Unpaid Earnings Today column table header")

includes/cli/class-affiliate-sub-commands.php
[L309]  __  Incorrect text domain used ("affiliate")
Warning: 23 problems Use --force to continue.
```

`class-metabox-overview-most-valuable.php` comes through clean, and the warnings above all look correct, the majority are where the `_e` function is being used with three arguments. 

So it looks to me that the original error report you posted is correct. 

Also, note that the line numbers and function name in the report all correspond to `class-metabox-overview-recent-referral-visits.php`, not `class-metabox-overview-most-valuable.php`.

**Comment by pippinsplugins** [comment]
> > > the majority are where the _e function is being used with three arguments.

Good catch. We were using `_ex` but then switched erroneously to `_e`. I've now corrected it back to `_ex`.

`grunt checktextdomain` still fails on `_ex` however. `checktextdomain` is reporting those lines as using `_e` with three arguments, but those lines are (now) actually using `_ex`.

Fixed function used: https://github.com/AffiliateWP/AffiliateWP/commit/c9daef55c45af797ee92ab71b617069082436fb3

New results of `checktextdomain`:

```
Pippins-MacBook-Air:affiliate-wp pippinwilliamson$ grunt checktextdomain
Running "checktextdomain:files" (checktextdomain) task

build/AffiliateWP/includes/admin/overview/metaboxes/class-metabox-overview-recent-referral-visits.php
[L66]  _e  Incorrect text domain used ("Affiliate column table header")
[L67]  _e  Incorrect text domain used ("URL column table header")
[L68]  _e  Incorrect text domain used ("Converted column table header")

build/AffiliateWP/includes/admin/overview/metaboxes/class-metabox-overview-recent-referrals.php
[L69]  _e  Incorrect text domain used ("Affiliate column table header")
[L70]  _e  Incorrect text domain used ("Amount column table header")
[L71]  _e  Incorrect text domain used ("Description column table header")

build/AffiliateWP/includes/admin/overview/metaboxes/class-metabox-overview-registrations.php
[L67]  _e  Incorrect text domain used ("Affiliate column table header")
[L68]  _e  Incorrect text domain used ("Status column table header")
[L69]  _e  Incorrect text domain used ("Actions column table header")

build/AffiliateWP/includes/admin/overview/metaboxes/class-metabox-overview-totals.php
[L54]   _e  Incorrect text domain used ("Paid Earnings column table header")
[L55]   _e  Incorrect text domain used ("Paid Earnings This Month column table header")
[L56]   _e  Incorrect text domain used ("Paid Earnings Today column table header")
[L80]   _e  Incorrect text domain used ("Unpaid Referrals column table header")
[L81]   _e  Incorrect text domain used ("Unpaid Referrals This Month column table header")
[L82]   _e  Incorrect text domain used ("Unpaid Referrals Today column table header")
[L105]  _e  Incorrect text domain used ("Unpaid Earnings column table header")
[L106]  _e  Incorrect text domain used ("Unpaid Earnings This Month")
[L107]  _e  Incorrect text domain used ("Unpaid Earnings Today column table header")
Warning: 18 problems Use --force to continue.

Aborted due to warnings.
```

**Comment by stephenharris** [comment]
I just checked out that the repo (at c9daef5) and it looks like all warnings are fixed. I've noticed that the files generating the errors above are in the `build` directory - could it be that it is out of sync with your source files? On another note, it looks like you're trying to [exclude the build directory](https://github.com/AffiliateWP/AffiliateWP/blob/f2d55a37d3ea5ea93c146c8a62e3f274b7d6ac1c/grunt/checktextdomain.js#L32) from that task - but evidently it isn't (perhaps it should be `'!build/**'`?).

If you're still convinced there is a bug then could you create a failing unit-test? Current tests are passing and [include `_ex`](https://github.com/stephenharris/grunt-checktextdomain/blob/48cb669ec3299679f308ed3ca23ae1db8f18a78d/test/fixtures/correct-domain.php#L6)

**Comment by pippinsplugins** [comment]
Ah! I didn't even notice it was looking at the build directory. My bad.

All good then!

**Comment by pippinsplugins** [comment]
Good catch on the builder directory exclusion, your change works. Thanks!
